examples/koans/drivable-speed.flix

```diff
@@ -0,0 +1,7 @@
+pub def drivable(roads: Array[(city, Int, city)], source: city, destination: city, minimumSpeed: Int): Bool  with Boxable[city] =
```

Comment:
I am surprised this function is able to be typed as pure! How?

Reply:
I was thinking about that too. Creating an array is impure right? but querying and projecting is pure even though the operate on arrays. Maybe this should be changed? It is nice that queries are pure though but to actually be pure it should probably be lists all around.